File: Numerical/spline/plot.py

# ==============================================================================
# Python Import Session
# ==============================================================================
import matplotlib.pyplot as plt
import seaborn
# ==============================================================================
# Receive Sequence
# ==============================================================================

temp = open("Data/Cubic.csv")
List = [line.split(',') for line in temp]
X, Y = [], []
for i, elem in enumerate(List):
    X.append(elem[0])
    Y.append(elem[1])
# ==============================================================================
# Plotting
# ==============================================================================
plt.figure(figsize=(10,10), dpi=600)
plt.plot(X, Y, label='Spline')
plt.title("Cubic")
plt.xlabel('x')
plt.ylabel('y')
plt.legend()
plt.savefig('Fig/Cubic.png')

File: Tutorial/numeric/plot.jl
using Winston

Data = readcsv("Test.csv")

X = Data[:,1]
Y = Data[:,2]

x = collect(-5.:1.:5.)
y = x.^2 +2x .- 3

F = FramedPlot(
    title="Test",
    xlabel="x",
    ylabel="y"
)

P = Points(x, y, size=.1)
C = Curve(X, Y)
setattr(P, "label", "points")
setattr(C, "label", "spline")
lgnd = Legend(.5, .8, [P, C])

add(F, P, C, lgnd)
savefig(F, "Test.svg", (1000,800))

run(`inkscape -z Test.svg -e Test.png -d 300 --export-background=WHITE`)
run(`rm Test.svg`)
